src/lib.rs

```diff
@@ -86,59 +86,43 @@ pub fn parse_recv_msg(buf: &[u8]) -> Sk8brdMsg {
     msg
 }
 
-pub fn console_print(buf: &[u8], len: u16) {
-    print!("{}", String::from_utf8_lossy(&buf[..len as usize]));
+pub fn console_print(buf: &[u8]) {
+    print!("{}", String::from_utf8_lossy(buf));
     stdout().flush().unwrap();
 }
 
 #[allow(clippy::explicit_write)]
-pub fn send_image(write_sink: &mut impl Write, buf: &[u8]) {
+pub fn send_image(write_sink: &mut impl Write, buf: &[u8]) -> Result<(), std::io::Error> {
     let mut last_percent_done: usize = 0;
     let mut bytes_sent = 0;
 
-    while bytes_sent < buf.len() {
-        let bytes_left = min(2048, buf.len() - bytes_sent);
+    for chunk in buf.chunks(2048) {
         let percent_done = 100 * bytes_sent / buf.len();
 
         if percent_done != last_percent_done {
             writeln!(stdout(), " Sending image: {}%\r", percent_done).unwrap();
         }
 
-        send_msg(
-            write_sink,
-            Sk8brdMsgs::MsgFastbootDownload,
-            bytes_left,
-            &buf[bytes_sent..],
-        );
+        send_msg(write_sink, Sk8brdMsgs::MsgFastbootDownload, chunk)?;
 
-        bytes_sent += bytes_left;
+        bytes_sent += chunk.len();
         last_percent_done = percent_done;
     }
 
     send_ack(write_sink, Sk8brdMsgs::MsgFastbootDownload)
 }
 
-pub fn select_brd(write_sink: &mut impl Write, name: &str) {
-    send_msg(
-        write_sink,
-        Sk8brdMsgs::MsgSelectBoard,
-        name.len(),
-        name.as_bytes(),
-    )
+pub fn select_brd(write_sink: &mut impl Write, name: &str) -> Result<(), std::io::Error> {
+    send_msg(write_sink, Sk8brdMsgs::MsgSelectBoard, name.as_bytes())
```

Comment:
Thanks to the message length strings are _not_ null-terminated? Note that Rust `str` allows inner `\0` characters.

Reply:
Yes. Rust has Pascal-like strings (IYKWYM). Also, note that I just removed the len from buf+len params.

Reply:
> Also, note that I just removed the len from buf+len params.

Yeah, it's always a bit tricky when looking at a... certain kind... of original codebase when the diff on top is only attempting to make things better.